plugins/deep-link/src/lib.rs

```diff
@@ -216,17 +188,31 @@ mod imp {
         ///
         /// This function updates the [`Self::get_current`] value and emits a `deep-link://new-url` event.
         #[cfg(desktop)]
-        pub fn handle_cli_arguments<S: AsRef<str>, I: Iterator<Item = S>>(&self, args: I) {
+        pub fn handle_cli_arguments<S: AsRef<str>, I: Iterator<Item = S>>(&self, mut args: I) {
             use tauri::Emitter;
 
             let Some(config) = &self.config else {
                 return;
             };
 
-            if let Some(url) = deep_link_from_args(config, args) {
-                let mut current = self.current.lock().unwrap();
-                current.replace(vec![url.clone()]);
-                let _ = self.app.emit("deep-link://new-url", vec![url]);
+            if cfg!(windows) || cfg!(target_os = "linux") {
+                args.next(); // bin name
+                let arg = args.next();
+
+                let maybe_deep_link = args.next().is_none(); // single argument
+                if !maybe_deep_link {
+                    return;
+                }
+
+                if let Some(url) = arg.and_then(|arg| arg.as_ref().parse::<url::Url>().ok()) {
+                    if config.desktop.contains_scheme(&url.scheme().to_string()) {
+                        let mut current = self.current.lock().unwrap();
+                        current.replace(vec![url.clone()]);
+                        let _ = self.app.emit("deep-link://new-url", vec![url]);
+                    } else if cfg!(debug_assertions) {
+                        log::warn!("argument {url} does not match any configured deep link scheme; skipping it");
+                    }
```

Comment:
do we really need to warn! here, shouldn't we just skip it?

Reply:
i think it's still important in case the dev is using runtime schemes and isn't aware of why it isn't being picked up for missing a note in the docs